packages/@react-spectrum/color/src/ColorWheel.tsx

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {classNames, dimensionValue} from '@react-spectrum/utils';
+import {ColorThumb} from './ColorThumb';
+import React, {useEffect, useRef, useState} from 'react';
+import {SpectrumColorWheelProps} from '@react-types/color';
+import styles from '@adobe/spectrum-css-temp/components/colorwheel/vars.css';
+import {useColorWheel} from '@react-aria/color';
+import {useColorWheelState} from '@react-stately/color';
+import {useFocus, useFocusVisible} from '@react-aria/interactions';
+import {useId} from '@react-aria/utils';
+import {useProviderProps} from '@react-spectrum/provider';
+
+const SEGMENTS = [];
+for (let i = 0; i < 360; i++) {
+  SEGMENTS.push(<rect width="80" height="2" x="80" y="79" fill={`hsl(${i}, 100%, 50%)`} transform={`rotate(${i} 80 80)`} key={i} />);
+}
+
+const RATIO_INNER_OUTER = 56 / 80;
+
+function ColorWheel(props: SpectrumColorWheelProps) {
+  props = useProviderProps(props);
+
+  let {isDisabled} = props;
+  let size = props.size && dimensionValue(props.size);
+
+  let inputRef = useRef(null);
+  let containerRef = useRef(null);
+
+  let [wheelRadius, setWheelRadius] = useState<number | null>(null);
+
+  useEffect(() => {
+    if (containerRef.current) {
+      setWheelRadius(containerRef.current.offsetWidth / 2);
+    }
+  }, [containerRef]);
```

Comment:
This is to get the size of the wheel if the `size` prop is used. 

Should this be a resize observer?

Reply:
So the `useResizeObserver` fires after every `useMove` event, not sure if this is bad for performance.

Is `size={'10vw'}` supported? Otherwise 
```js
  useEffect(() => {
    if (containerRef.current) {
      setWheelRadius(containerRef.current.offsetWidth / 2);
    }
  }, [size, containerRef]);
```
would work as well...

Reply:
I suppose it could be a percentage potentially (e.g. to fill a container `size="100%"`).

`useResizeObserver` shouldn't fire unless it actually resized... Maybe you need to make your `onResize` function memoized with `useCallback` so it doesn't create a new resize observer on every render?